cp_diffdock_service.py

```diff
@@ -0,0 +1,188 @@
+import asyncio
+import json
+import logging
+import re
+import os
+from websockets.server import serve
+import concurrent.futures
+from argparse import ArgumentParser
+
+from cp_diffdock_protocol import DiffDockProtocol
+from cp_diffdock_api import DiffDockApi
+from cp_ws_helpers import extractWsAppMessage, formWsAppMessage, formCompletedMessage
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger('diffdock_service')
+
+#
+# This is a websocket server for servicing DiffDock requests.
+# This listens on a port and places incoming json requests into a queue, serviced by a number of workers.
+#
+
+async def handleRequest(websocket, queue):
+    """
+    Receive a request and put it into the queue.
+    Make sure we keep the websocket so we can send the response when docking finishes.
+    """
+    try:
+        message = await websocket.recv()
+    except Exception as ex:
+        log.error(f"Exception reading from websocket: {ex}. Can't handle this request.")
+        return
+
+    log.info("Received a request.")
+    # Optional special handling for messages from bfd-server or other Bioleap WsApps
+    # requestId is non-empty if it is from a wsApp, and will need to be included in responses.
+    requestId, cmdName, requestData = extractWsAppMessage(message)
+    requestContext = websocket, requestId
+
+    # Parse
+    try:
+        requestObj = DiffDockProtocol.Request.from_json(requestData)
+    except:
```

Comment:
Some linter I have used (not sure if it's flake8) slapped my wrist for unspecific except. It wanted more like `except Exception as ex:`. I suppose there's little point in logging a stack trace from an exception from_json but maybe the exception text would be heplful.

Reply:
log.exception includes the stacktrace